src/Misc/textureTools.ts

```diff
@@ -72,4 +73,77 @@ export class TextureTools {
 
         return rtt;
     }
+
+    /**
+     * Reads the pixels stored in the webgl texture and returns them as a base64 string
+     * @param texture defines the texture to read pixels from
+     * @param faceIndex defines the face of the texture to read (in case of cube texture)
+     * @param level defines the LOD level of the texture to read (in case of Mip Maps)
+     * @returns The base64 encoded string or null
+     */
+    public static GenerateBase64StringFromTexture(texture: Texture, faceIndex = 0, level = 0): Nullable<string> {
+
+        var internalTexture = texture.getInternalTexture();
+        if (!internalTexture || internalTexture.isCube) {
+            return null;
+        }
+
+        var pixels = texture.readPixels(faceIndex, level);
+        if (!pixels) {
+            return null;
+        }
+
+        var size = texture.getSize();
+        var width = size.width;
+        var height = size.height;
+
+        if (pixels instanceof Float32Array) {
+            var len = pixels.byteLength / pixels.BYTES_PER_ELEMENT;
+            var npixels = new Uint8Array(len);
+
+            while (--len >= 0) {
+                var val = pixels[len];
+                if (val < 0) {
+                    val = 0;
+                } else if (val > 1) {
+                    val = 1;
+                }
+                npixels[len] = val * 255;
+            }
+
+            pixels = npixels;
+        }
+
+        var canvas = document.createElement('canvas');
+        canvas.width = width;
+        canvas.height = height;
+
+        var ctx = canvas.getContext('2d');
+        if (!ctx) {
+            return null;
+        }
+
+        var imageData = ctx.createImageData(width, height);
+        imageData.data.set(pixels);

```

Comment:
@deltakosh 
I don't know what it wants :)
Pixels is always a Uint8Array at this point in the code, it has a length property..
Any idea?

    [at-loader] ../../src/Misc/textureTools.ts:127:28 
    TS2345: Argument of type 'ArrayBufferView' is not assignable to parameter of type 'ArrayLike<number>'.
    Property 'length' is missing in type 'ArrayBufferView' but required in type 'ArrayLike<number>'.

Reply:
you should be able to cast it

Reply:
Okay, that should fix it.
Now there's just the linting left, it started again with circular dependencies.
TextureTools import Texture, Texture import TextureTools.. thoughts?

Reply:
we can create a new file then like CopyTools

Reply:
I'll have it finished by tomorrow at the latest
Thanks for the guidance both @deltakosh & @Popov72